cmd/worker/main.go

```diff
@@ -87,12 +89,31 @@ func saveResults(ctx context.Context, pkg *pkgmanager.Pkg, dest resultBucketPath
 			return fmt.Errorf("failed to upload static analysis results to blobstore = %w", err)
 		}
 	}
+
+	fileWriteDataUploadStart := time.Now()
 	if dest.fileWrites != "" {
-		err := resultstore.New(dest.fileWrites, resultstore.ConstructPath()).Save(ctx, pkg, dynamicResults.FileWrites)
-		if err != nil {
+		rs := resultstore.New(dest.fileWrites, resultstore.ConstructPath())
```

Comment:
This code is duplicated in `cmd/analyze/main.go`, is it possible to extract it into a function in `internal/worker`?